lambda-handler.js

```diff
@@ -31,11 +31,19 @@ exports.handler = function(event, context, callback) {
 			}
 			return context.fail(err);
 		}
+		// If this is a signed URL, we need to calculate the max-age of the image.
+		let maxAge = 31536000;
+		if ( args['X-Amz-Expires'] ) {
+			// Date format of X-Amz-Date is YYYYMMDDTHHMMSSZ, which is not parsable by Date.
+			const date = new Date( args['X-Amz-Date'].replace(/(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})Z/, '$1-$2-$3T$4:$5:$6Z' ) );
```

Comment:
This would be a bit nicer if it were split out into two steps I think, given it's a long line.